cmd/block-generator/generator/generate.go

```diff
@@ -218,13 +210,14 @@ func (g *generator) generatePaymentTxn(round uint64, intra uint64) (types.Signed
 	receiver := indexToAccount(receiveIndex)
 
 	amount := g.balances[sendIndex] / 2
-	g.balances[sendIndex] -= amount
-	g.balances[receiveIndex] += amount
 
-	if g.balances[sendIndex] < 0 {
-		panic(fmt.Sprintf("the balance fell below zero for idx %d", sendIndex))
+	if g.balances[sendIndex] < amount {
```

Comment:
This obviously cannot happen. Why are you adding this?

Reply:
I changed the comparison based on your feedback, it is checking for < amount to detect an underflow.

Reply:
a < a/2 is always false

Reply:
What I'm trying to do with this check is make sure the program exits if the account balance is too small to make a useful transaction. I see now that the logic isn't correct. I guess I could `g.balances[sendIndex]` > 1, which would also avoid the possible divide-by-zero crash.

This code is going to have to change anyway since it doesn't currently support closing an account. So I don't think this needs to be fixed.

Reply:
Ok, just remove the check.